src/ast-utils.js

```diff
@@ -10,6 +10,9 @@ export function isReference(node, parent) {
 	// disregard the `bar` in `export { foo as bar }`
 	if (parent.type === 'ExportSpecifier' && node !== parent.local) return false;
 
+	// disregard assigned references
+	if (parent.type === 'AssignmentExpression' && parent.left === node) return false;
+
```

Comment:
I see the problem but I would avoid putting the exception here as this basically makes `isReference` unusable for any other reference detection. As a matter of fact, I would propose to use the chance to remove `isReference` from the bundle in favour of the `is-reference` package that is used by Rollup itself, is better maintained and handles more edge cases properly (such as

```js
require:
while(true) {
  break require;
}
```

Instead, I would just put the exception into `transform.js`:

```js
if (isReference(node, parent) && !scope.contains(node.name) && !(parent.type === 'AssignmentExpression' && parent.left === node)) { /* ... */ }
```